lib/spack/spack/util/executable.py

```diff
@@ -269,17 +278,21 @@ def which_string(*args, **kwargs):
     if isinstance(path, string_types):
         path = path.split(os.pathsep)
 
+
+
     for name in args:
-        if sys.platform == "win32":
```

Comment:
This chunk of code is probably still needed. The executable name passed to which will be the same for both linux and windows, and won't contain the '.exe' extension. The tests will need to be updated too.